tests/phpunit/tests/admin/wpUpgrader.php

```diff
@@ -0,0 +1,1656 @@
+	 *
+	 * @ticket 54245
+	 *
+	 * @covers WP_Upgrader::release_lock
+	 */
+	public function test_release_lock_should_remove_lock_option() {
+		global $wpdb;
+
+		$this->assertSame(
+			1,
+			$wpdb->insert(
+				$wpdb->options,
+				array(
+					'option_name'  => 'lock.lock',
+					'option_value' => 'content',
+				),
+				'%s'
+			),
+			'The initial lock was not created.'
+		);
+
+		WP_Upgrader::release_lock( 'lock' );
+
+		$this->assertNotSame( 'content', get_option( 'lock.lock' ) );
+	}
+
+	/**
+	 * Tests that `WP_Upgrader::download_package()` returns early when
+	 * the 'upgrader_pre_download' filter returns a non-false value.
+	 *
+	 * @ticket 54245
+	 *
+	 * @covers WP_Upgrader::download_package
+	 */
+	public function test_download_package_should_exit_early_when_the_upgrader_pre_download_filter_returns_non_false() {
+		self::$upgrader_skin_mock->expects( $this->never() )->method( 'feedback' );
+
+		add_filter(
+			'upgrader_pre_download',
+			static function () {
+				return 'a non-false value';
+			}
+		);
+
+		$result = self::$instance->download_package( 'package' );
+
+		$this->assertSame( 'a non-false value', $result );
+	}
+
+	/**
+	 * Tests that `WP_Upgrader::download_package()` should apply
+	 * 'upgrader_pre_download' filters with expected arguments.
+	 *
+	 * @ticket 54245
+	 *
+	 * @covers WP_Upgrader::download_package
+	 */
+	public function test_download_package_should_apply_upgrader_pre_download_filter_with_arguments() {
+		self::$upgrader_skin_mock->expects( $this->never() )->method( 'feedback' );
+
+		add_filter(
+			'upgrader_pre_download',
+			function ( $reply, $package, $upgrader, $hook_extra ) {
+				$this->assertFalse( $reply, '"$reply" was not false' );
+
+				$this->assertSame(
+					'package',
+					$package,
+					'The package file name was not "package"'
+				);
+
+				$this->assertSame(
+					self::$instance,
+					$upgrader,
+					'The wrong WP_Upgrader instance was passed'
+				);
+
+				$this->assertSameSets(
+					array( 'hook_extra' ),
+					$hook_extra,
+					'The "$hook_extra" array was not the expected array'
+				);
+
+				return ! $reply;
+			},
+			10,
+			4
+		);
+
+		$result = self::$instance->download_package( 'package', false, array( 'hook_extra' ) );
+
+		$this->assertTrue(
+			$result,
+			'WP_Upgrader::download_package() did not return true'
+		);
+	}
+
+	/**
+	 * Tests that `WP_Upgrader::download_package()` returns an existing file.
+	 *
+	 * @ticket 54245
+	 *
+	 * @covers WP_Upgrader::download_package
+	 */
+	public function test_download_package_should_return_an_existing_file() {
+		$result = self::$instance->download_package( __FILE__ );
+
+		$this->assertSame( __FILE__, $result );
+	}
+
+	/**
+	 * Tests that `WP_Upgrader::download_package()` returns a WP_Error object
+	 * for an empty package.
+	 *
+	 * @ticket 54245
+	 *
+	 * @covers WP_Upgrader::download_package
+	 */
+	public function test_download_package_should_return_a_wp_error_object_for_an_empty_package() {
```

Comment:
Nuke this test for this PR as it requires the source change which will only be in 6.5+

Reply:
@costdev tests relating to the src changes have been removed in https://github.com/WordPress/wordpress-develop/pull/5556/commits/e5685d1beda7c09b7fe7e566b999bd67a8c9f38f